app/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnector.scala

```diff
@@ -96,48 +110,64 @@ class HelpToSaveProxyConnectorImpl @Inject() (http:              WSHttp,
     )
   }
 
-  override def getAccount(nino: String, queryString: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[AccountO] = {
+  override def getAccount(nino: String, systemId: String, correlationId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[Option[Account]] = {
 
-    val url = s"$proxyURL/help-to-save-proxy/nsi-services/account?$queryString"
+    val url = s"$proxyURL/help-to-save-proxy/nsi-services/account?nino=$nino&correlationId=$correlationId&version=$getAccountVersion&systemId=$systemId"
     val timerContext = metrics.getAccountTimer.time()
-    EitherT[Future, String, AccountO](
-      http.get(url).map[Either[String, AccountO]] {
+    EitherT[Future, String, Option[Account]](
+      http.get(url).map[Either[String, Option[Account]]] {
         response ⇒
-          val time = timerContext.stop()
-          val correlationId = "correlationId" -> getCorrelationId(queryString)
+          val _ = timerContext.stop()
           response.status match {
             case Status.OK ⇒
-              val result = response.parseJson[NsiAccount].map(a ⇒ AccountO(Account(a)))
+              val result = response.parseJson[NsiAccount].flatMap(a ⇒
+                Account(a).toEither.bimap(s ⇒ s"[${s.toList.mkString(",")}]", Some(_)))
 
               result.fold(
                 e ⇒ {
-                  logger.warn(s"Could not parse getNsiAccount response, received 200 (OK), error=$e, response = ${maskNino(response.body)}", nino, correlationId)
+                  logger.warn(s"Could not parse getNsiAccount response, received 200 (OK), error=$e, response = ${maskNino(response.body)}",
```

Comment:
I don't think the response from NS&I should be logged to Kibana, even with the NINO masked, because it could contain other PII such as the individual's name & address.

Reply:
I'm thinking of a situation where NS&I return a success response but we can't parse it (for example because some of the fields are missing or in the wrong format) - would this line be executed in that situation?